### ISSUE TITLE
Add support for removing "data" and "datacontenttype" from CloudEventBuilder

### ISSUE THREAD

**Issue by johanhaleby** [cloudevents/sdk-java#373]
Hi!

I want to duplicate a CloudEvent but remove its "data" and "datacontenttype". I've tried to do:

```java
var originalCloudEvent = ... 
CloudEvent eventWithoutData = new CloudEventBuilder(originalCloudEvent).withData((byte[]) null).withDataContentType(null).build();
```

But this fails with:

```
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:222)
	at io.cloudevents.core.data.BytesCloudEventData.<init>(BytesCloudEventData.java:20)
	at io.cloudevents.core.data.BytesCloudEventData.wrap(BytesCloudEventData.java:54)
	at io.cloudevents.core.impl.BaseCloudEventBuilder.withData(BaseCloudEventBuilder.java:66)
```

Now I need to clone the `originalCloudEvent` manually which is suboptimal. I'd like methods similar to `withoutExtension` but for "data" and "datacontenttype". WDYT?

**Comment by slinkydeveloper** [comment]
:+1: for adding it, would you mind doing a PR? :smile:

**Comment by johanhaleby** [comment]
@slinkydeveloper I'm trying to create a PR but I'm confused. If I just clone the project and create my branch from master then I can find the `CloudEventBuilder` interface. But if I follow the instructions in https://github.com/cloudevents/sdk-java/blob/master/pr_guidelines.md the `CloudEventBuilder` interface seems to be removed? And I cannot find the `withoutExtension` method anywhere.

**Comment by slinkydeveloper** [comment]
@johanhaleby are you sure you're not cloning the 1.x branch?

**Comment by johanhaleby** [comment]
@slinkydeveloper Could be, but I'm following the instructions in  https://github.com/cloudevents/sdk-java/blob/master/pr_guidelines.md :) But let me try again.

**Comment by johanhaleby** [comment]
@slinkydeveloper I removed my previous git repo and now I think it works :) https://github.com/cloudevents/sdk-java/pull/374
